src/main/java/picard/sam/UmiMetrics.java

```diff
@@ -81,10 +100,67 @@ public UmiMetrics(final int length, final int observedUniqueUmis, final int infe
         OBSERVED_BASE_ERRORS = observedBaseErrors;
         DUPLICATE_SETS_WITHOUT_UMI = duplicateSetsWithoutUmi;
         DUPLICATE_SETS_WITH_UMI = duplicateSetsWithUmi;
-        EFFECTIVE_LENGTH_OF_INFERRED_UMIS = effectiveLengthOfInferredUmis;
-        EFFECTIVE_LENGTH_OF_OBSERVED_UMIS = effectiveLengthOfObservedUmis;
-        ESTIMATED_BASE_QUALITY_OF_UMIS = estimatedBaseQualityOfUmis;
-        EXPECTED_READS_WITH_UMI_COLLISION = expectedUmiCollisions;
+        INFERRED_UMI_ENTROPY = effectiveLengthOfInferredUmis;
+        OBSERVED_UMI_ENTROPY = effectiveLengthOfObservedUmis;
+        UMI_BASE_QUALITIES = estimatedBaseQualityOfUmis;
+        UMI_COLLISION_EST = expectedUmiCollisions;
         UMI_COLLISION_Q = umiCollisionQ;
     }
+
+    public void calculateDerivedFields() {
+        OBSERVED_UNIQUE_UMIS = observedUmis.size();
+        INFERRED_UNIQUE_UMIS = inferredUmis.size();
+
+        OBSERVED_UMI_ENTROPY = effectiveNumberOfBases(observedUmis);
+        INFERRED_UMI_ENTROPY = effectiveNumberOfBases(inferredUmis);
+
+        UMI_COLLISION_Q = -10 * Math.log10(UMI_COLLISION_EST / inferredUmis.size());
+        estimateBaseQualities(observedUmiBases);
+    }
+
+    private double effectiveNumberOfBases(Histogram<?> observations) {
+        double entropyBase4 = 0.0;
+
+        double totalObservations = observations.getSumOfValues();
+        for (Histogram.Bin observation : observations.values()) {
+            double pObservation = observation.getValue() / totalObservations;
+            entropyBase4 = entropyBase4 - pObservation * Math.log(pObservation);
+        }
+
+        // Convert to log base 4 so that the entropy is now a measure
+        // of the effective number of DNA bases.  If we used log(2.0)
+        // our result would be in bits.
+        return entropyBase4 / Math.log(4.0);
+    }
+
+    public void updateDuplicateSetMetrics(final int maxEditDistanceToJoin, final int duplicateSetSize) {
```

Comment:
a public function needs a java-doc.. what is this function. what does it do? what are the inputs?